### PR TITLE
fix: use local nickname in chat input link previews

### PR DESCRIPTION
**Pull request by igor-sirotin**
Fixes https://github.com/status-im/status-desktop/issues/12624

image

### CODE REVIEW

**Review by caybro — APPROVED** [review]
LGTM

**Review by alexjba — APPROVED** [review]
LGTM! 👍 

Minor question inline

**Review by kounkou — APPROVED** [review]

File: src/app/modules/main/chat_section/chat_content/input_area/view.nim [hunk 1]
```diff
@@ -238,6 +238,11 @@ QtObject:
   proc updateLinkPreviewsFromCache*(self: View, urls: seq[string]) =
     let linkPreviews = self.delegate.linkPreviewsFromCache(urls)
     self.linkPreviewModel.updateLinkPreviews(linkPreviews)
+    
+    for contactId in self.linkPreviewModel.getContactIds().items:
+      let contact = self.delegate.getContactDetails(contactId)
+      if contact.dto.displayName != "":
```
[the remark this answers is not in the document]
Comment (kounkou) [reply src/app/modules/main/chat_section/chat_content/input_area/view.nim:244]: nit. For faster performance, wouldn't it be better to first fetch `self.linkPreviewModel.getContactIds().items` in a variable and then iterate on the variable instead?

[the remark this answers is not in the document]
Comment (igor-sirotin) [reply src/app/modules/main/chat_section/chat_content/input_area/view.nim:244]: > Not sure I understand why we need the `if` statement. In which case the `displayName` is empty?

When for some reason we don't have information about the contact in this cache. The cache never returns a `nil` value, but returns an empty contact instead.

[the remark this answers is not in the document]
Comment (igor-sirotin) [reply src/app/modules/main/chat_section/chat_content/input_area/view.nim:244]: > nit. For faster performance, wouldn't it be better to first fetch `self.linkPreviewModel.getContactIds().items` in a variable and then iterate on the variable instead?

I think it shouldn't affect performance. This is done as in a default nim example for `items` 🙂

**Pull request merged**